pymeasure/instruments/hp/hp856Xx.py

```diff
@@ -0,0 +1,3239 @@
+
+        Type: :code:`float`
+
+        .. code-block:: python
+
+            # set the demodulation time to 1.2 seconds
+            instr.demodulation_time = 1.2
+
+            if instr.demodulation_time == 10:
+                pass
+
+        """,
+        validator=strict_range,
+        values=[100e-3, 60],
+    )
+
+    detector_mode = Instrument.control(
+        "DET?", "DET %s",
+        """
+        Control the IF detector used for acquiring measurement data.
+        This is normally a coupled function, in which the spectrum analyzer selects the
+        appropriate detector mode. Four modes are available: normal, positive, negative, and sample.
+
+        Type: :code:`str`
+
+        Takes a representation of the detector mode, either from :class:`DetectionModes` or
+        use 'NEG', 'NRM', 'POS', 'SMP'
+
+        .. code-block:: python
+
+            instr.detector_mode = DetectionModes.SMP
+            instr.detector_mode = 'NEG'
+
+            if instr.detector_mode == DetectionModes.SMP:
+                pass
+
+        """,
+        validator=strict_discrete_set,
+        values=[e for e in DetectionModes]
+    )
+
+    # now implemented as a property but due to the ability of the underlying gpib command to
+    # specify the unit, there would be an alternative implementation as a method to allow the user
+    # to modify the setting unit without manipulating it via the 'amplitude_unit' property
+    display_line = Instrument.control(
+        "DL?", "DL %s",
+        """
+        Control the horizontal display line for use as a visual aid or for
+        computational purposes. The default value is 0 dBm.
+
+        Type: :code:`float`, :code:`str`
+
+        Takes a value with the unit of :attr:`amplitude_unit` or 'ON' / 'OFF'
+
+        .. code-block:: python
+
+            instr.display_line = 'ON'
+            instr.display_line = -10
+
+            if instr.detector_mode == 0:
+                pass
+
+        """,
+        validator=joined_validators(strict_range, strict_discrete_set),
+        values=[[float("-inf"), float("inf")], ["ON", "OFF"]],
+        set_process=lambda v: v if isinstance(v, str) else '%.11E' % v
+    )
+
+    done = Instrument.measurement(
+        "DONE?",
+        """
+        Get back (e.g. return) when all commands in a command string
+        entered before 'done' has been completed. Sending a :meth:`trigger_sweep` command
+        before 'done' ensures that the spectrum analyzer will complete a full sweep before
+        continuing on in a program.
+        Depending on the timeout a timeout error from the adapter will raise before the spectrum
+        analyzer can finish due to an extreme long sweep time
+
+        .. code-block:: python
+
+            instr.trigger_sweep()
+
+            # wait for a full sweep and than 'do_something'
+            if instr.done:
+                do_something()
+
+        """,
+        map_values=True,
+        values={True: "1"},
+        cast=str
+    )
```

Comment:
As you do not return anything (or does it return something different than "1"?), this should be better a method:
```suggestion
    def check_done(self):
        """Check whether the device has finished all commands previously sent."""
        answer = self.ask("DONE?")
        # TODO do here some error checking on the answer?
```